src/main/scala/za/co/absa/commons/s3/S3Location.scala

```diff
@@ -49,10 +53,28 @@ object SimpleS3Location {
     }
 
     def isValidS3Path: Boolean = S3LocationRx.pattern.matcher(path).matches
-  }
 
-  def apply(path: String): SimpleS3Location = {
-    path.toSimpleS3Location.getOrElse(throw new IllegalArgumentException(s"Could not parse S3 location from $path!"))
+    def ensureS3LocationEndsWithoutSlash: SimpleS3Location = {
+      val parsedS3Location = apply(path)
+
+      if (parsedS3Location.path.endsWith("/")) parsedS3Location.copy(path=parsedS3Location.path.dropRight(1))
+      else parsedS3Location
+    }
+
+    def ensureS3LocationEndsWithSlash: SimpleS3Location = {
+      val parsedS3Location = apply(path)
```

Comment:
```suggestion
      val parsedS3Location = SimpleS3Location(path)
```
The more clarity the better